srtcore/api.cpp

```diff
@@ -370,10 +465,21 @@ int CUDTUnited::newConnection(const SRTSOCKET listen, const sockaddr_any& peer,
       return -1;
    }
 
-   enterCS(m_IDLock);
-   ns->m_SocketID = -- m_SocketIDGenerator;
-   HLOGF(mglog.Debug, "newConnection: generated socket id %d", ns->m_SocketID);
-   leaveCS(m_IDLock);
+   try
+   {
+       CGuard l_idlock(m_IDLock);
+       ns->m_SocketID = generateSocketID();
+   }
+   catch (CUDTException& e)
```

Comment:
Why don't you catch the exception in the above call to `generateSocketID()` then?

Reply:
Because the function that calls the `newSocket` function will catch it anyway.

The `CUDT::newConnection` OTOH will be called asynchronically as a part of the handshake process, therefore no other part will catch it.

Everything is explained in the comments.

Reply:
OTOH you might be right in one thing - if this throws an exception, no one will clean the allocated object. I have to put everything under the try block then.

Reply:
Well, no. There would be a different kind of exception to be thrown. I need to add an interceptor.